entity-service-impl/src/main/java/org/hypertrace/entity/query/service/EntityQueryServiceImpl.java

```diff
@@ -89,14 +91,37 @@ public void execute(EntityQueryRequest request, StreamObserver<ResultSetChunk> r
         .convertToEDSQuery(request, attrNameToEDSAttrMap.get(request.getEntityType()));
     Iterator<Document> documentIterator = entitiesCollection.search(
         DocStoreConverter.transform(tenantId.get(), query));
-    List<Entity> entities = convertDocsToEntities(documentIterator);
-
-    //Build result
-    //TODO : chunk response. For now sending everything in one chunk
-    responseObserver.onNext(convertEntitiesToResultSetChunk(
-        entities,
-        request.getSelectionList(),
-        attrNameToEDSAttrMap.get(request.getEntityType())));
+
+    boolean isFirstEntity = true;
+    int chunkId = 0;
+    while (documentIterator.hasNext()) {
```

Comment:
What if `documentIterator.hasNext()` is empty, what would be the return response? I think we were sending only metadata headers in ResultSetChunk but no rows if search results were empty - `entitiesCollection.search`. I think we should also do the same here.

Reply:
Makes sense, for backward compatibility of clients